### DISCUSSION TITLE
patch2self question

### DISCUSSION THREAD

**Discussion by araikes** [dipy/dipy#2917] [Preprocessing]
Hi @ShreyasFadnavis,

I'm using patch2self on ex-vivo mouse data. I was reading the documentation where it says this:

"Patch2Self preserved anatomical detail. This can be visually verified by inspecting the residuals obtained above. Since we do not see any structure in the difference residuals, it is clear that it preserved the underlying signal structure and got rid of the stochastic noise.".

I'm looking at my data and, for any given volume, there is some measure of structure and if I take the average of all of the `rms_diff` volumes, I see a brain. Does this mean that patch2self is not functioning as expected for these data or am I just overthinking it? Screenshot below is `rms_diff` for a selected volume out of 89 (left) and the mean `rms_diff` across all 89 volumes.

![image](https://github.com/dipy/dipy/assets/18663624/aefa0058-24e6-462c-a03b-3ee70262d97e)

**Comment by skoudoro** [comment]
@ShreyasFadnavis ?? Thank you for asking @araikes

**Comment by araikes** [comment]
@ShreyasFadnavis bump on this

**Comment by ShreyasFadnavis** [comment]
Hi @araikes -- sorry for being late on this! I think it can be okay, what's the dimensionality on your data? Increasing the patch_radius=[1, 1, 1] may help, computationally feasible.

**Comment by araikes** [comment]
The data are 200 microns (isotropic) with a 64 x 96 x 64 grid size

**Reply by ShreyasFadnavis** [reply]
Since the data is not that big, I recommend increasing the resolution. Let me know if it helps!

**Reply by araikes** [reply]
Do you mean upsample the data or change the patch radius?

**Reply by ShreyasFadnavis** [reply]
I meant change the patch radius!

**Comment by skoudoro** [comment]
related to #3976 and should be fixed. closing
